phoenix-core/src/main/java/org/apache/phoenix/jdbc/LoggingPhoenixPreparedStatement.java

```diff
@@ -45,7 +45,9 @@ public ResultSet executeQuery() throws SQLException {
 
     @Override
     public ResultSet getResultSet() throws SQLException {
-        return new LoggingPhoenixResultSet(super.getResultSet(), phoenixMetricsLog, sql);
+        ResultSet resultSet = super.getResultSet();
+        return (resultSet == null) ? null : new LoggingPhoenixResultSet(super.getResultSet(),
```

Comment:
This is still a bug. `super.getResultSet()` call is not idempotent. You should replace this with `resultSet` local variable